libs/core/algorithms/include/hpx/parallel/unseq/simd_helpers.hpp

```diff
@@ -0,0 +1,163 @@
+            HPX_PRAGMA_VECTOR_UNALIGNED HPX_VECTOR_REDUCTION(| : found_flag)
+            for (IterDiff j = i; j < i + num_blocks; ++j)
+            {
+                std::int32_t const t = f(*(first + j));
+                simd_lane[j - i] = t;
+                found_flag |= t;
+            }
+            // clang-format on
+
+            if (found_flag)
+            {
+                IterDiff j;
+                for (j = 0; j < num_blocks; ++j)
+                {
+                    if (simd_lane[j])
+                    {
+                        break;
+                    }
+                }
+                return first + i + j;
+            }
+            i += num_blocks;
+        }
+
+        //Keep remainder scalar
+        while (i != n)
+        {
+            if (f(*(first + i)))
+            {
+                return first + i;
+            }
+            ++i;
+        }
+        return first + i;
+#endif    //HPX_EARLYEXIT_PRESENT
+    }
+
+    template <typename Iter1, typename Iter2, typename IterDiff, typename F>
+    std::pair<Iter1, Iter2> unseq2_first_n(Iter1 const first1,
+        Iter2 const first2, IterDiff const n, F&& f) noexcept
+    {
+#if HPX_EARLYEXIT_PRESENT
+        IterDiff i = 0;
+
+        // clang-format off
+        HPX_PRAGMA_VECTOR_UNALIGNED HPX_PRAGMA_SIMD_EARLYEXIT
+        for (; i < n; ++i)
+            if (f(*(first1 + i), *(first2 + i)))
+                break;
+        // clang-format on
+
+        return std::make_pair(first1 + i, first2 + i);
+#else
+        Iter1 const last1 = first1 + n;
+        Iter2 const last2 = first2 + n;
+
+        static constexpr std::int32_t num_blocks =
+            HPX_LANE_SIZE / sizeof(std::int32_t);
+        alignas(HPX_LANE_SIZE) std::int32_t simd_lane[num_blocks] = {0};
+
+        std::size_t outer_loop_ind = 0;
+        while (outer_loop_ind <= n - num_blocks)
+        {
+            std::int32_t found_flag = 0;
+            IterDiff i;
+
+            // clang-format off
+            HPX_PRAGMA_VECTOR_UNALIGNED HPX_VECTOR_REDUCTION(| : found_flag)
+            for (i = 0; i < num_blocks; ++i)
+            {
+                IterDiff const t = f(*(first1 + outer_loop_ind + i),
+                    *(first2 + outer_loop_ind + i));
+                simd_lane[i] = t;
+                found_flag |= t;
+            }
+            // clang-format on
+
+            if (found_flag)
+            {
+                IterDiff i2;
+                for (i2 = 0; i2 < num_blocks; ++i2)
+                {
+                    if (simd_lane[i2])
+                        break;
+                }
+                return std::make_pair(
+                    first1 + outer_loop_ind + i2, first2 + outer_loop_ind + i2);
+            }
+            outer_loop_ind += num_blocks;
+        }
+
+        //Keep remainder scalar
+        for (; outer_loop_ind != n; ++outer_loop_ind)
+            if ((f, *(first1 + outer_loop_ind), *(first2 + outer_loop_ind)))
```

Comment:
Does this realy do what you want? Shouldn't this be:
```suggestion
            if (f(*(first1 + outer_loop_ind), *(first2 + outer_loop_ind)))
```